### ISSUE TITLE
Batch matching results by donor metadata and HLA type

### ISSUE THREAD

**Issue by zabeen** [Anthony-Nolan/Atlas#916]
This ticket is dependent on changes made as part of https://github.com/Anthony-Nolan/Atlas/issues/897.

Search orchestrator [downloads all matching results](https://github.com/Anthony-Nolan/Atlas/blob/826995232bb2caa04f5defc8b54b057da2478a68/Atlas.Functions/DurableFunctions/Search/Activity/SearchActivityFunctions.cs#L68), and then [groups donors by HLA tying and metadata](https://github.com/Anthony-Nolan/Atlas/blob/94808497f9fd9efadaa7d06f2d23e51274fe79ee/Atlas.MatchPrediction/ExternalInterface/DonorInputBatcher.cs#L21) to generate match prediction requests, as an optimisation to minimise the total number of requests.

Where results batching is enabled, if matching results are also batched in the same way (by donor typing and metadata), then result blobs would not have to be all downloaded to memory before match prediction requests are generated, which would flatten the memory usage curve.

**Comment by zabeen** [comment]
Blocked by merge of #897, and also by performance testing to see if this change is actually needed.

**Comment by zabeen** [comment]
There is a better way to deal with the problem of downloading large result sets prior to match prediction. Will explain on another ticket when I get back from A/L, but closing this for now. Also, testing shows memory exception occurring later in search (#931) so the priority is to fix that step.
